include/mxnet/base.h

```diff
@@ -307,6 +312,21 @@ inline Context Context::GPU(int32_t dev_id) {
   return Create(kGPU, dev_id);
 }
 
+inline int32_t Context::GetGPUCount() {
+#if MXNET_USE_CUDA
+  int32_t count;
+  cudaError_t e = cudaGetDeviceCount(&count);
+  if (e == cudaErrorNoDevice) {
+    return 0;
+  }
+  CHECK_EQ(e, cudaSuccess) << " CUDA: " << cudaGetErrorString(e);
+  return count;
+#else
+  LOG(FATAL) << "Please compile with CUDA support to query the number of GPUs.";
```

Comment:
I think it is valid to do this call since the frontend apis do not know whether the underlying library was compiled with or without CUDA support. Instead, we should also have a second API which provides that information.